central/cve/datastoretest/datastore_sac_test.go

```diff
@@ -0,0 +1,1239 @@
+			genericStore: genericCVEStore,
+		}
+		s.nodeCVEStore = &nodeCVEDataStoreFromGenericStore{
+			genericStore: genericCVEStore,
+		}
+	}
+	s.imageTestContexts = sacTestUtils.GetNamespaceScopedTestContexts(context.Background(), s.T(), resources.Image)
+	s.nodeTestContexts = sacTestUtils.GetNamespaceScopedTestContexts(context.Background(), s.T(), resources.Node)
+}
+
+func (s *cveDataStoreSACTestSuite) TearDownSuite() {
+	s.Require().NoError(s.dackboxTestStore.Cleanup())
+}
+
+func (s *cveDataStoreSACTestSuite) getCVEID(cve string) string {
+	if features.PostgresDatastore.Enabled() {
+		return cve + "#Linux"
+	}
+	return cve
+}
+
+func (s *cveDataStoreSACTestSuite) cleanImageToVulnerabilitiesGraph() {
+	s.Require().NoError(s.dackboxTestStore.CleanImageToVulnerabilitiesGraph())
+}
+
+func (s *cveDataStoreSACTestSuite) cleanNodeToVulnerabilitiesGraph() {
+	s.Require().NoError(s.dackboxTestStore.CleanNodeToVulnerabilitiesGraph())
+}
+
+type existsTestCase struct {
+	contextKey     string
+	expectedExists bool
+}
+
+type getTestCase struct {
+	contextKey    string
+	expectedFound bool
+}
+
+type readMultiTestCase struct {
+	contextKey      string
+	expectedFetched []string
+}
+
+type countTestCase struct {
+	contextKey    string
+	expectedCount int
+}
+
+func (s *cveDataStoreSACTestSuite) TestSACImageCVEExistsSingleScopeOnly() {
+	// Inject the fixture graph, and test exists for CVE-1234-0001
+	s.dackboxTestStore.PushImageToVulnerabilitiesGraph()
+	defer s.cleanImageToVulnerabilitiesGraph()
+	cveId := s.getCVEID("CVE-1234-0001")
+	testCases := map[string]existsTestCase{
+		"Unrestricted read-write can see single scope CVE": {
+			contextKey:     sacTestUtils.UnrestrictedReadWriteCtx,
+			expectedExists: true,
+		},
+		"Unrestricted read can see single scope CVE": {
+			contextKey:     sacTestUtils.UnrestrictedReadCtx,
+			expectedExists: true,
+		},
+		"Right cluster full read-write can see single scope CVE": {
+			contextKey:     sacTestUtils.Cluster1ReadWriteCtx,
+			expectedExists: true,
+		},
+		"Right cluster partial read-write to right namespace can see single scope CVE": {
+			contextKey:     sacTestUtils.Cluster1NamespaceAReadWriteCtx,
+			expectedExists: true,
+		},
+		"Right cluster partial read-write to wrong namespace can NOT see single scope CVE": {
+			contextKey: sacTestUtils.Cluster1NamespaceBReadWriteCtx,
+		},
+		"Right cluster partial read-write to at least the right namespace can see single scope CVE": {
+			contextKey:     sacTestUtils.Cluster1NamespacesABReadWriteCtx,
+			expectedExists: true,
+		},
+		"Wrong cluster full read-write can NOT see single scope CVE": {
+			contextKey: sacTestUtils.Cluster2ReadWriteCtx,
+		},
+		"Wrong cluster partial read-write to the matching namespace can NOT see single scope CVE": {
+			contextKey: sacTestUtils.Cluster2NamespaceAReadWriteCtx,
+		},
+		"Wrong cluster partial read-write to any not-matching namespace can NOT see single scope CVE": {
```

Comment:
Potentially room for one more test case:
`Wrong cluster partial read-write to at least the right namespace can NOT see single scope CVE`